sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -526,9 +531,9 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
    * @param maxFields Maximum number of fields that will be converted to strings.
    *                  Any elements beyond the limit will be dropped.
    */
-  def simpleString(maxFields: Int): String = {
-    s"$nodeName ${argString(maxFields)}".trim
-  }
+  def simpleString(maxFields: Int): String = s"$nodeName ${argString(maxFields)}".trim
+
+  def simpleStringWithNodeId(): String = simpleString(0)
```

Comment:
I'd rather leave it abstract, to make sure it's implemented by `QueryPlan` and we implement `Expression.simpleStringWithNodeId` by throwing exception.

Reply:
@cloud-fan Good idea Wenchen. Thanks.